spec/xrootd-cmstfc.spec

```diff
@@ -11,9 +10,28 @@ URL: https://github.com/bbockelm/xrootd-cmstfc
 # git-archive master | gzip -7 > ~/rpmbuild/SOURCES/xrootd-lcmaps.tar.gz
 Source0: %{name}.tar.gz
 BuildRoot: %(mktemp -ud %{_tmppath}/%{name}-%{version}-%{release}-XXXXXX)
-BuildRequires: xrootd-libs-devel xerces-c-devel pcre-devel
+
+%define xrootd_current_major 4
```

Comment:
Should we allow this to be overridden via an RPM macro specified at build time?

Reply:
I'm not sure how this would play with osg-build, could we leave that for the next time?

Reply:
```
if 0%{?osg}
%define xrootd_current_major 4
%endif
```

This should be sufficient (assuming this looks right to @matyasselmeci)

Reply:
I thought that the idea was to be able to set the version number at build time

Reply:
I think what @brianhlin wrote will work. I'm not sure what you mean by "RPM macro specified at build time".

Reply:
> I thought that the idea was to be able to set the version number at build time

This gets us halfway there: if someone wants to build this outside of an OSG build environment, they'll have to specify `xrootd_current_major`. @matyasselmeci is there a way to specify `xrootd_current_major` as an argument to `osg-build`?

Reply:
Passing arbitrary --define options to rpmbuild and mock commands is doable but not currently implemented. You can currently work around it by defining those in your `~/.rpmmacros` file.

Passing --defines to Koji builds is impossible.

I strongly caution against depending on users to define specific macros via the command line, since it harms reproducibility -- you have to remember what arguments you passed to rpmbuild. If you want to change something, edit the spec file.

Reply:
It would just be nice to specify a single spec file that can support builds against XRootD 4 and XRootD 5. This will have to do, though.